#                             Parkinson - Disease -Classifier

## Let Us Import some libraries



In [93]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score
from sklearn import metrics
from sklearn.linear_model import LogisticRegression


#### Data
Data is read from file which was created from dat_file.py using data from the dataset

In [94]:
data = pd.read_csv("file.csv")


In [95]:
data

pressure   grip_angle    timestamp    state  status
0   884.200931   794.317300   115.650000   static       1
1   910.966791   662.641217   115.650000  dynamic       1
2   788.450502   782.556671  1052.316667   static       1
3   926.910511   902.801802  1052.316667  dynamic       1
4   843.550013   903.502152  1228.116667   static       1
..         ...          ...          ...      ...     ...
75  520.355333  1060.966411  1171.950000  dynamic       0
76  899.069890  1474.577348  1307.450000   static       0
77  899.257785  1432.905643  1307.450000  dynamic       0
78  607.163567   879.065934  1417.933333   static       0
79  409.578713   941.002075  1417.933333  dynamic       0

[80 rows x 5 columns]

lets split the data using in built functions from the sklearn lib 

In [96]:
# Let us now split the data into train and test before it shuffle the data

X_train, X_test, y_train, y_test = train_test_split(data, data.status,test_size=0.2, random_state=42, shuffle=True)


##### Print the shapes of training and test datasets

In [97]:
print(X_train.shape)
print(y_train.shape)
print(X_test.shape)
print(y_test.shape)


(64, 5)
(64,)
(16, 5)
(16,)


## Random Forest Classifier
 ###### Let's add the 4 Features from the file  and train our  model  

In [98]:
'''
RANDOM FOREST MODEL FOR CLASSIFICATION
'''

features = ["pressure","grip_angle","timestamp","state"]

X = pd.get_dummies(X_train[features])
X_test = pd.get_dummies(X_test[features])

model = RandomForestClassifier(n_estimators = 50,max_depth = 4 ,random_state=0 ) 
model.fit(X,y_train)
y_pred = model.predict(X_test)
print("Accuracy of Random Forest:",metrics.accuracy_score(y_test, y_pred))


Accuracy of Random Forest: 0.875


Bingo! we have got an accuracy of 0.875
Tune the hyperparameters to get more accuracy

# Logistic Regression

In [99]:
'''
LOGISTIC REGRESSION MODEL FOR CLASSIFICATION
'''
model2 = LogisticRegression(penalty= 'none' ,random_state=0 ,max_iter=100,solver='lbfgs').fit(X,y_train)
y_pred2 = model2.predict(X_test)

print("Accuracy of Logistic regression:",metrics.accuracy_score(y_test, y_pred2))

Accuracy of Logistic regression: 0.875


Note: This file is still to be updated